docs/Authenticating-Users-via-LDAP.md

```diff
@@ -0,0 +1,110 @@
+
+## Why is LDAP Relevant to cBioPortal?
+
+The cBioPortal code has no means of storing user name and passwords and no means of directly authenticating users.  If you want to restrict access to your instance of cBioPortal, you therefore have to consider an external authentication service. LDAP is one means of doing so, and your larger institution may already provide LDAP support. This greatly simplifies user management.
+
+
+# Configuring LDAP within cBioPortal
+
+
+## Modifying portal.properties
+
+Within portal.properties, modify the section labeled `authentication`.  For example:
+
+    # authentication
+    authenticate=ldap
+    
+    ## configuration for the LDAP access
+    ldap.user_search_base=DC=example,DC=com
+    ldap.url=ldap://ldap.example.com:389
+    ldap.manager.dn=CN=manager-user,DC=example,DC=com
+    ldap.manager.password=PASSWORD
+    ## The following attributes for are good for ActiveDirectory, for OpenLDAP use "uid" for username
+    ldap.attributes.username=sAMAccountName
+    #ldap.attributes.username=uid
+    ldap.attributes.last_name=sn
+    ldap.attributes.given_name=givenName
+    ldap.attributes.email=mail    
+
+Please note that you will have to modify all the above to match your own settings.
+
+## Authorizing Users
+
+Within portal.properties, modify
+
+    authorization=true
+
+Next, please read the Wiki page on [User Authorization](User-Authorization.md), and add user rights for a single user.
+
+
+## Configuring the Login.jsp Page
+
+The login page is configurable via the `portal.properties` properties `skin.authorization_message`. 
+For example in `skin.authorization_message` you can be set to something like this:
+
+```
+skin.authorization_message= Welcome to this portal. Access to this portal is available to authorized test users at YOUR ORG.  [<a href="http://thehyve.nl/">Request Access</a>].
+```
+
+You can also set a standard text in `skin.login.contact_html` that will appear in case of problems: 
+
+```
+skin.login.contact_html=If you think you have received this message in error, please contact us at <a style="color:#FF0000" href="mailto:cbioportal-access@your.org">cbioportal-access@your.org</a>
+```
+
+
+## Doing a Test Run
+
+You are now ready to go.
+
+Rebuild the WAR file and re-deploy:
+
+```
+mvn -DskipTests clean install
+cp portal/target/cbioportal.war $CATALINA_HOME/webapps/
+```
+
+Then, go to:  [http://localhost:8080/cbioportal/](http://localhost:8080/cbioportal/).
+
+If all goes well, the following should happen:
+
+* You will be redirected to the Login Page.
+* After authenticating, you will be redirected the home page.
+
+Example of Login page:
+
+![Login page example](images/previews/login_page_example.png)
+
+If this does not happen, see the Troubleshooting Tips  below.
+
+
+## Troubleshooting Tips 
+
+### Logging
```

Comment:
One thing you can add here to troubleshoot security related errors:

log4j.logger.org.springframework.security=DEBUG